### PR TITLE
Modeling users

### PR DESCRIPTION
**Pull request by stratospherique**

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
```
Comment (rvvergara) [inline app/models/comment.rb]: It would be great to include a validation for comment body presence :)

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    t.index ["user_id"], name: "index_comments_on_user_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.string "title"
+    t.text "body"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "user_id"
+    t.index ["user_id"], name: "index_posts_on_user_id"
```
Comment (rvvergara) [inline db/schema.rb]: Let's add foreign_key constraints for `user_id`.

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ActiveRecord::Schema.define(version: 20190517232013) do
+
+  create_table "comments", force: :cascade do |t|
+    t.text "body"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "post_id"
+    t.integer "user_id"
+    t.index ["post_id"], name: "index_comments_on_post_id"
+    t.index ["user_id"], name: "index_comments_on_user_id"
```
Comment (rvvergara) [inline db/schema.rb]: Let's add foreign_key constraints for `user_id` and for `post_id`

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :posts
```
Comment (rvvergara) [inline app/models/user.rb]: While in this case we aren't required to do so, it's still good to get used to adding `dependent :destroy` into your `has_many` associations. For instance in this case we normally would want to remove all `user` 's posts and comments if the user's account is deleted

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
```
Comment (rvvergara) [inline app/models/post.rb]: Let's add a `dependent :destroy` here.

File: app/models/comment.rb (hunk 1 above)
Comment (jofer86) [reply app/models/comment.rb]: Validation for presence is set to true.

File: app/models/user.rb (hunk 4 above)
Comment (jofer86) [reply app/models/user.rb]: Added the dependant :destroy to the has many assosiation.

File: app/models/post.rb (hunk 5 above)
Comment (jofer86) [reply app/models/post.rb]: Added dependant: :destroy to belongs to and has many assosiations here.

File: db/schema.rb (hunk 3 above)
Comment (jofer86) [reply db/schema.rb]: Replace add reference with add foreign key.

File: db/schema.rb (hunk 2 above)
Comment (jofer86) [reply db/schema.rb]: Replace add reference with add foreign key.

**Comment by stratospherique** [comment]
In our version of rails (5.1.6), adding a foreign key using active records methods actually adds a foreign key constraint silently (i.e not showing in the schema.rb). there is an github issue discussing this [here](https://github.com/rails/rails/issues/35207)

**Pull request merged**